src/order/category/NonemptyFinLinOrd.lean

```diff
@@ -80,6 +82,83 @@ equivalence.mk dual dual
   (nat_iso.of_components (λ X, iso.mk $ order_iso.dual_dual X) $ λ X Y f, rfl)
   (nat_iso.of_components (λ X, iso.mk $ order_iso.dual_dual X) $ λ X Y f, rfl)
 
+lemma epi_iff_surjective {A B : NonemptyFinLinOrd.{u}} (f : A ⟶ B) :
+  epi f ↔ function.surjective f :=
+begin
+/- This proof is similar to that of `simplex_category.epi_iff_surjective`. -/
```

Comment:
Do you think you can deduplicate parts?

Reply:
In my original draft code, I was deducing the case of `NonemptyFinLinOrd` from that of `simplex_category`, but it would introduce weird dependencies: in order to get `epi_iff_surjective` for `NonemptyFinLinOrd`, we would have to import `simplex_category`. Maybe I can try to do it the other way: keep the proof of `epi_iff_surjective` for `NonemptyFinLinOrd` and deduce the result for `simplex_category` (which imports `NonemptyFinLinOrd` anyway), and similarly for `mono_iff_surjective`?

Reply:
I have just made this change. `mono_iff_injective` and `epi_if_surjective` are now first established for `NonemptyFinLinOrd` and the similar results for `simplex_category` are deduced from these.